static/scripts/index.js

```diff
@@ -0,0 +1,513 @@
+        retrieveDataLocally();
+        let comments = retrieveCellsMetaLocally();
+        let isCollectedMetaData = true;
+
+        const table = new Handsontable(example, {
+            data: data,
+            fixedColumnsLeft: lenFixedHeaders,
+            fixedRowsTop: lenFixedRowsText,
+            manualColumnMove: true,
+            manualColumnResize: true,
+            manualRowResize: true,
+            mergeCells: mergeCellsSettings,
+            comments: true,
+            hiddenColumns: {
+                columns: [0],
+                indicators: true,
+            },
+            hiddenRows: {
+                rows: [0],
+                indicators: true,
+            },
+            contextMenu: ['commentsAddEdit', 'commentsRemove', 'hidden_columns_hide', 'hidden_rows_hide', 'hidden_columns_show', 'hidden_rows_show'],
+            filters: true,
+            dropdownMenu: ['filter_by_value', 'filter_action_bar', 'undo'],
+            className: 'controlsQuickFilter htCenter htMiddle',
+            colHeaders: allHeaders,
+            columns: columns,
+            colWidths: 100,
+            columnHeaderHeight: 225,
+            rowHeaders: fixedRowsText,
+            rowHeaderWidth: 125,
+            init: function () {
+                if (comments) {
+                    comments.forEach(comment => {
+                        this.setCellMeta(comment.row, comment.col, comment.key, comment.value);
+                    });
+                }
+                isCollectedMetaData = false;
+            },
+            afterGetColHeader: function (col, th) {
+                th.style.transform = 'rotate(180deg)';
+                th.style.writingMode = 'vertical-lr';
+                th.style.textAlign = 'center';
+                th.style.fontWeight = 'bold';
+                th.style.height = '100px';
+                th.style.lineHeight = '100px';
+
+                if (col > lenFixedHeaders - 1) {
+                    let colData = this.getDataAtCol(col);
+                    if (colData[5] >= colData[4]) {
+                        th.style.backgroundColor = 'green';
+                    }
+                }
+            },
+            afterChange: function (changes) {
+                if (changes) {
+                    changes.forEach(([row, prop, oldValue, newValue]) => {
+                        let col = this.propToCol(prop);
+                        if ((col >= lenFixedHeaders && row >= lenFixedRowsText) && (row % 2 === 1)) {
+                            //Boolean to determine whether a cell is updated or empty
+                            let isDeleted = false;
+
+                            //Update total course assistants
+                            if (oldValue !== null && newValue === null) {
+                                isDeleted = true;
+                            }
+
+                            const colInfos = this.getDataAtProp(prop);
+                            let nbrAssistants = colInfos[5];
```

Comment:
I realize that specific column indexes are put in the code as is. Maybe it would be more maintenable to have an enum with the column name and indexes somewhere to specify the column name instead of its index. 

This would ease column addition/deletion in the future as well updating the code if a column position is changed.